### ISSUE TITLE
Date params conversion

### ISSUE THREAD

**Issue by pankleks** [brianc/node-postgres#1172]
Hi,

I setup node env with TZ=UTC
I run update statement with date params - dates are in UTC ("2016-11-28T08:00:00.000Z")
Still my date is converted to computer local time.
Does pg respects TZ setting of node env?

Thanks!

**Comment by cressie176** [comment]
I think you have to set it on your database session. 

https://www.postgresql.org/docs/9.1/static/sql-set.html
```
SET [ SESSION | LOCAL ] TIME ZONE { timezone | LOCAL | DEFAULT }
```

**Comment by pankleks** [comment]
Thanks, but that's not it - DB is set to UTC.
It's explicitly pg-node that converts dates:(
I would be nice to have someone explain exactly how pg-node is  handling this.

**Comment by ikokostya** [comment]
@pankleks `node-postgres` uses [node-pg-types](https://github.com/brianc/node-pg-types) for parsing postgres types, which internally uses [postgres-data](https://github.com/bendrucker/postgres-date) for parsing dates. 

Value which represents `timestamp without timezone`  postgres type doesn't contain offset (even `Z`). Thus `postgres-data` parses this datetime using javascript [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) constructor as [local time](https://github.com/bendrucker/postgres-date/blob/6e23dbc767cc0cea70a67a04e0e682ed0b9cd676/index.js#L37-L44). You can read more about this behavior in ECMAScript specification https://tc39.github.io/ecma262/#sec-date-time-string-format

> When the time zone offset is absent, date-only forms are interpreted as a UTC time and date-time forms are interpreted as a local time.

To resolve this issue, you can override default parser for datetime and parse it always in `UTC` format.

**Comment by ikokostya** [comment]
I agree that this behavior isn't obvious. So, I suggest to describe it in README. Even better to add markdown table with type correspondence like in `rust-postgres`: https://github.com/sfackler/rust-postgres#type-correspondence

**Comment by guikubivan** [comment]
To be explicit, do something similar to this post: https://github.com/brianc/node-postgres/issues/1746#issuecomment-435678889
